cmd/system-probe/api/client/client.go

```diff
@@ -81,7 +82,36 @@ func constructURL(module string, endpoint string) string {
 	return u.String()
 }
 
+// URL constructs a system-probe URL for a module-less endpoint.
+func URL(endpoint string) string {
+	return constructURL("", endpoint)
+}
+
+// DebugURL constructs a system-probe URL for the debug module and endpoint.
+func DebugURL(endpoint string) string {
+	return constructURL("debug", endpoint)
+}
+
 // ModuleURL constructs a system-probe ModuleURL given the specified module and endpoint.
 func ModuleURL(module types.ModuleName, endpoint string) string {
 	return constructURL(string(module), endpoint)
 }
+
+// ReadAllResponseBody reads the entire HTTP response body as a byte slice
+func ReadAllResponseBody(resp *http.Response) ([]byte, error) {
+	// if we are not able to determine the content length
+	// we read the whole body without pre-allocation
+	if resp.ContentLength <= 0 {
+		return io.ReadAll(resp.Body)
```

Comment:
should we introduce some limit here to avoid "memory bombs"? do we fully control the response or rather its size can be affected by "external" parameters?

Reply:
I consider that out of scope for this PR, because I'm just moving an existing function. https://github.com/DataDog/datadog-agent/blob/dccac0b0bce2aa41708f4b2ddb7d37f1784df159/pkg/process/net/common.go#L495-L512